flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSource.java

```diff
@@ -176,23 +179,47 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
                     && filters.size() > 0) {
                 ((BulkDecodingFormat<RowData>) bulkReaderFormat).applyFilters(filters);
             }
-            BulkFormat<RowData, FileSourceSplit> bulkFormat =
+
+            BulkFormat<RowData, FileSourceSplit> format =
```

Comment:
Creating a method for that is not trivial because of the different return type of the `DecodingFormat`, I've refactored with `if/else`, I hope it helps